File: math/prob728.py
# 728 Self Dividing Numbers

class Solution(object):
    def selfDividingNumbers(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: List[int]
        """
        ans = [ n  for n in range(left, right+1) if self.test(n)]
        return ans

    def test(self, n):
        """if n is self divisible"""
        for c in str(n):
            d = int(c)
            if d == 0:
                return False
            elif n % d:
                return False
        return True

File: math/prob728.kt
// 728 Brute force 5% solution
class Solution {
    fun selfDividingNumbers(left: Int, right: Int): List<Int> {
        
        val ans = mutableListOf<Int>()
        for (v in left..right) {
            if (helper(v)){
                ans.add(v)
            }
        }
    
        return ans
    }
    
    fun helper(n: Int): Boolean {
        var s = n.toString().toList().toSet().map {it-> it.toString().toInt()}
        if (0 in s) {
            return false
        } else {
            for (d in s) {
                if (n % d != 0) {
                    return false
                }
            }
            return true
        }
    }

}
